spring-integration-zip/src/main/java/org/springframework/integration/zip/transformer/UnZipTransformer.java

```diff
@@ -131,6 +133,16 @@ public void process(InputStream zipEntryInputStream, ZipEntry zipEntry) throws I
 							tempDir.mkdirs(); //NOSONAR false positive
 							final File destinationFile = new File(tempDir, zipEntryName);
 
+							/* If we see the relative traversal string of ".." we need to make sure
+							 * that the outputdir + name doesn't leave the outputdir.
+							 */
+							if (zipEntryName.contains("..") &&
+									!destinationFile.getCanonicalPath().startsWith(workDirectory.getCanonicalPath())) {
+								throw new ZipException("The file " + zipEntryName +
+										" is trying to leave the target output directory of " + workDirectory
+										+ ". Ignoring this file.");
```

Comment:
Here, we're not just ignoring this file, we're rejecting the entire zip file (which is probably the best thing to do). Just needs a fix to the exception message, to remove the last sentence.

Reply:
Yes, please.